_posts/2024-01-16-fido-passkey.md

```diff
@@ -0,0 +1,1127 @@
+Users often reuse the same password across multiple platforms, fail to rotate passwords, or use weak ones. Have you explored the website '[have i been pwned?](https://haveibeenpwned.com/){:target="_blank" rel="noopener noreferrer"}'? 
+Take a look; you'll be surprised at how many of your passwords have already been leaked.
+
+It appears to be high time to address the vulnerabilities associated with traditional username and password logins and transition towards the use of FIDO.
+
+FIDO (Fast IDentity Online) is a set of open authentication standards that aims to replace traditional passwords with more secure and convenient methods.
+
+Key benefits of FIDO authentication:
+* Strong Security: FIDO Authentication utilises public-key cryptography, which is more secure than traditional passwords against phishing attacks, keyloggers, and other threats.
+* Convenience: Passkeys, the credential type enabled by FIDO Authentication, are stored on your device and can be unlocked using biometrics like fingerprint or face recognition, eliminating the need to remember or type passwords repeatedly.
+
+## How does it work
+
+To understand how passkeys work, let's take a step back and delve into the realm of asymmetric cryptography. In asymmetric cryptography, 
+we deal with a key pair consisting of a private key, which must remain private and undisclosed, and a public key, which can be shared openly.
+
+Through asymmetric encryption, a message encrypted with the public key can only be deciphered by the possessor of the private key. 
+Conversely, if a message is encrypted with the private key, it can be read using the public key (signature).
+
+So, how is this technology employed in FIDO? During registration, your device generates a unique passkey, which contains a public key and a private key.
+The public key is stored on the service provider's server, while the private key remains securely on your device (smartphone, tablet, or a dedicated FIDO security key).
+
+To register as a new user, we must send our public key to the server and have it validated. 
+The server will send us a challenge that we must sign using our private key. 
+If the server can verify the signature using our public key, the registration is complete.
+
+To authenticate a user, we send our username to the server, and the server will answer with a challenge encrypted with our public key.
+On our device, we need to decrypt the challenge with our private key, solve the challenge, and return our response encrypted with our private key.
+After sending the encrypted response, the server is able to validate the response with our public key and check if the login succeeded.
+
+So, in this way, we no longer store the user's password, but only a public key that can be shared openly.
+
```

Comment:
Maybe add a small diagram demonstrating the flow with messages going back and forth? Or link to a good YouTube video with decent visuals.